src/main/fc/tasks.c

```diff
@@ -113,11 +113,32 @@
 
 static void taskMain(timeUs_t currentTimeUs)
 {
-    UNUSED(currentTimeUs);
-
 #ifdef USE_SDCARD
     afatfs_poll();
 #endif
+
+#ifdef USE_DSHOT_TELEMETRY
+    static bool dshotTelemetryActive = false;
+    static timeUs_t nextDshoTelemetryCheck = 0;
+
+    if (!ARMING_FLAG(ARMED)) {
+        if (motorConfig()->dev.useDshotTelemetry && !dshotTelemetryActive && (currentTimeUs > nextDshoTelemetryCheck))  {
+            nextDshoTelemetryCheck = currentTimeUs + 1e6; // check again in 1 second
+            dshotTelemetryActive = true;
+            for (uint8_t i = 0; i < getMotorCount(); i++) {
```

Comment:
This loop appears twice, and should probably be extracted into a function, and potentially pushed down into the driver.

Reply:
Yeah, this is mostly an artifact of being based on #7813. Once that gets merged I'll rebase this and combine the ESC active telemetry checks.